Tweak the RD's suplexing ability into a skillchip

coiax opened the pull request:
:cl: coiax
tweak: The Research Director's knowledge of suplexing immovable rods is
now in a skillchip.
/:cl:

Although this doesn't change the power level of the Research Director,
it's important to rework "job title" checks into traits or something
similar.

Moving job title specific behaviour out of checks by job name and into
traits means there's more flexibility to trigger behaviour. If an admin
feels the station really needs the ability, they can var edit it in, or
spawn the chip.

coiax committed:
Tweak the RD's suplexing ability into a skillchip

:cl: coiax
tweak: The Research Director's knowledge of suplexing immovable rods is
now in a skillchip.
/:cl:

Although this doesn't change the power level of the Research Director,
it's important to rework "job title" checks into traits or something
similar.

Moving job title specific behaviour out of checks by job name and into
traits means there's more flexibility to trigger behaviour. If an admin
feels the station really needs the ability, they can var edit it in, or
spawn the chip.

coiax committed:
Improve skillchip name

A dot after a fake acroynym looks better.

ATH1909 commented on code/modules/jobs/job_types/research_director.dm:
```diff
@@ -51,6 +51,8 @@
 	backpack = /obj/item/storage/backpack/science
 	satchel = /obj/item/storage/backpack/satchel/tox
 
+	skillchips = list(/obj/item/skillchip/job/research_director)
```
we really ought to give RDs that roboticist skillchip that tells you which cyborg wire is which

coiax replied:
Yes, it would be out of scope. But it's an easy thing to do?

Fikou commented on code/modules/library/skill_learning/job_skillchips/research_director.dm:
```diff
@@ -0,0 +1,9 @@
+/obj/item/skillchip/job/research_director
+	name = "R.D.S.P.L.X skillchip"
```
there should be a dot after X

JohnFulpWillard commented:
Im not a fan of this, I dont like how a chip in your brain would give you superhuman strength capable of suplexing a rod, things like serial numbers and wires make sense because its mind alterations, but strength just doesnt fit that

san7890 commented:
> Im not a fan of this, I dont like how a chip in your brain would give you superhuman strength capable of suplexing a rod, things like serial numbers and wires make sense because its mind alterations, but strength just doesnt fit that

Maybe it's not raw strength that lets an RD take down a rod. Think about it, Research Directors are meant to be dorky geeks in stuffy beige suits and clipboards, yet they can stop an unstoppable rod. It makes more sense for it to be a technical thing that requires a skillset and knowledge, rather than brute force.

Rohesie commented on code/modules/events/immovable_rod.dm:
```diff
@@ -174,22 +174,20 @@ In my current plan for it, 'solid' will be defined as anything with density == 1
-				new /obj/structure/festivus/anchored(drop_location())
-				new /obj/effect/anomaly/flux(drop_location())
-				qdel(src)
+	if(HAS_TRAIT(user, TRAIT_ROD_SUPLEX) || (user.mind && HAS_TRAIT(user.mind, TRAIT_ROD_SUPLEX)))
```
```suggestion
	if(!HAS_TRAIT(user, TRAIT_ROD_SUPLEX) && !(user.mind && HAS_TRAIT(user.mind, TRAIT_ROD_SUPLEX)))
		return
```
Could you reduce this proc's horizontal drift with early returns?

Rohesie commented on code/modules/library/skill_learning/job_skillchips/research_director.dm:
```diff
@@ -0,0 +1,9 @@
+	skill_name = "True Strength"
+	skill_description = "The knowledge and strength to resolve the most ancient conumdrum; what happens when an unstoppable force meets an immovable object."
+	skill_icon = "dumbbell"
+	activate_message = "<span class='notice'>You feel the urge to prove yourself in the upcoming Feats of Strength.</span>"
+	deactivate_message = "<span class='notice'>You no care about the upcoming Feats of Strength, and are no longer clear on what they were in the first place.</span>"
```
Could you change this message to be a bit nerdier?
It's the Research Director, not an athlete.

coiax committed:
Merge branch 'master' into suplex-skillchip

coiax committed:
Add early return, tweak skillchip wording

Rohesie commented on code/modules/events/immovable_rod.dm:
```diff
@@ -174,22 +174,28 @@ In my current plan for it, 'solid' will be defined as anything with density == 1
+		new /obj/effect/anomaly/flux(drop_location())
+		qdel(src)
+
+	. = TRUE
```
```suggestion
	return TRUE
```

Rohesie commented on code/modules/events/immovable_rod.dm:
```diff
@@ -174,22 +174,28 @@ In my current plan for it, 'solid' will be defined as anything with density == 1
+
+	playsound(src, 'sound/effects/meteorimpact.ogg', 100, TRUE)
+	for(var/mob/M in urange(8, src))
+		if(!M.stat)
+			shake_camera(M, 2, 3)
```
```suggestion
		if(M.stat)
			continue
		shake_camera(M, 2, 3)
```

coiax committed:
Feedback from code review

- Invert an if loop inside a for loop
- Return explicitly rather than implicitly.
